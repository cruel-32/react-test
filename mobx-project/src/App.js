import React from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Home from 'pages/home';
import Event from 'pages/event';
import Login from 'pages/login';
import Join from 'pages/join';
import 'styles/App.scss';

// https://fonts.googleapis.com/css?family=Nanum+Gothic:400,700,800|Open+Sans:300,300i,400,400i,600,600i,700,700i,800,800i&amp;subset=korean'

class App extends React.Component {
  render(){
    return <Router>
      <div className="App">
        <div>
          <p>테스트 페이지. 코드보고 눌러도 보고 대충 감을 잡으셈들</p>
          <Link to={`/`}>Home</Link>
          <Link to={`/login`}>Login</Link>
          <Link to={`/join`}>Join</Link>
          <Link to={`/event/2000?count=10`}>Event</Link>

          <Route exact path="/" component={Home}/>
          <Route exact path="/login" component={Login} />
          <Route exact path="/join" component={Join} />
          <Route path={["/event/:id", "/event"]} component={Event} />

        </div>
      </div>
    </Router>
  }
}
export default App;
