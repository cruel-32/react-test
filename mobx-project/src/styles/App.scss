@import 'import/mixin.scss';

button {
  background:#ccc;
  color:$textBold;
}