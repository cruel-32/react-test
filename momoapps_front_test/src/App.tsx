import React, { useState, FC } from 'react';
import { BrowserRouter, Route, Link } from "react-router-dom";

import LoginPage from 'containers/login'
import AccountDetail from 'containers/accountDetail'
import AccountList from 'containers/accountList'

import { Button } from '@material-ui/core';

const App:FC = ()=> {
  const [mode, setMode] = useState('on');

  // return (
  //   <div>app</div>
  // )

  return (
    <BrowserRouter>
      <div className="App">
        <div>
          <Link to="/">home</Link>&nbsp;&nbsp;
          <Link to={`/account/${'tester'}`}>AccountDetail</Link>&nbsp;&nbsp;
          <Link to="/account?sort=name">list</Link>&nbsp;&nbsp;
        </div>
        <Route exact path='/'>
          <LoginPage source={mode} />
        </Route>
        <Route exact path='/account/:_id'>
          <AccountDetail source={mode}/>
        </Route>
        <Route exact path='/account'>
          <AccountList source={mode}></AccountList>
        </Route>
        <Button className="app__button--on" onClick={()=>{setMode('on')}}>set on</Button>
        <button className="app__button--off" onClick={()=>{setMode('off')}}>set off</button>
        
      </div>
    </BrowserRouter>
  );
}

export default App;
