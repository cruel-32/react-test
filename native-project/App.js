import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>Open up App.js to start working on your app</Text>
        <Text>안드로이드폰으로 리액트 네이티브 프로젝트</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
