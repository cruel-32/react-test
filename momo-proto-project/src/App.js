import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { login, logout } from './reducers/user'

class App extends Component{

  loginAction = (e) =>{
    const {login} = this.props;
    login({
      name:'test1',
      age:1
    });
  }

  logoutAction = () =>{
    const {logout} = this.props;
    logout();
  }


  render(){
    const {loggedUser} = this.props;
    console.log('loggedUser : ', loggedUser);

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          {loggedUser ? loggedUser.name : ''}
          <div onClick={this.loginAction}>
            로그인
          </div>
          <div onClick={this.logoutAction}>
            로그아웃
          </div>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

// props 로 넣어줄 스토어 상태값
const mapStateToProps = state => ({
  loggedUser: state.user.loggedUser,
});

// props 로 넣어줄 액션 생성함수
const mapDispatchToProps = {login,logout};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);