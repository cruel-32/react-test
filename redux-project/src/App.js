import React, { Component } from 'react';

import './App.css';
import WatingListContainer from './containers/WatingListContainer';
import PaletteContainer from './containers/PaletteContainer'; // **** (1) 불러오기
import CounterContainer from './containers/CounterContainer'; // **** (1) 불러오기
import MemberListContainer from './containers/MemberListContainer'; // **** (1) 불러오기
import UserListContainer from './containers/UserListContainer'; // **** (1) 불러오기

class App extends Component {
  render() {
    console.log('App mounted');
    return (
      <div className="App">
        <PaletteContainer />
        <CounterContainer />
        <WatingListContainer />
        <MemberListContainer />
        <UserListContainer />
      </div>
    );
  }
}

export default App;