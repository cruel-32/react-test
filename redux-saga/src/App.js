import React, { useState } from 'react';
import { BrowserRouter, Route, Link } from "react-router-dom";

import 'styles/App.scss';
import LoginPage from 'containers/login'
import AccountDetail from 'containers/accountDetail'
import AccountList from 'containers/accountList'

function App() {
  const [mode, setMode] = useState('on');

  return (
    <BrowserRouter>
      <div className="App">
        <div>
          <Link to="/">home</Link>&nbsp;&nbsp;
          <Link to={`/account/${'tester'}`}>AccountDetail</Link>&nbsp;&nbsp;
          <Link to="/account?sort=name">list</Link>&nbsp;&nbsp;
        </div>
        <Route exact path='/'>
          <LoginPage source={mode} />
        </Route>
        <Route exact path='/account/:_id'>
          <AccountDetail source={mode}/>
        </Route>
        <Route exact path='/account'>
          <AccountList source={mode}></AccountList>
        </Route>
        <button onClick={()=>{setMode('on')}}>set on</button>
        <button onClick={()=>{setMode('off')}}>set off</button>
        
      </div>
    </BrowserRouter>
  );
}

export default App;
