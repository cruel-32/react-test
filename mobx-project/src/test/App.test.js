import React from 'react';
import { configure, mount, shallow, render  } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from 'App';

configure({ adapter: new Adapter() });

describe('App은 잘 되고 있습니다', ()=>{
  it('생성과 삭제 이상없이 되고 있습니다.', () => {
    const wrapper = mount(<App/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('상태값은 없습니다', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.state()).toBe(null);
  });
});
