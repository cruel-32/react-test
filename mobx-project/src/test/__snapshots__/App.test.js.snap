// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App은 잘 되고 있습니다 상태값은 없습니다 1`] = `null`;

exports[`App은 잘 되고 있습니다 생성과 삭제 이상없이 되고 있습니다. 1`] = `ReactWrapper {}`;

exports[`App은 잘 되고 있습니다 생성과 삭제 이상없이 되고 있습니다. 2`] = `ReactWrapper {}`;
